tests/api.c

```diff
@@ -6863,6 +6871,1086 @@ static int test_wc_Arc4Process (void)
 
 }/* END test_wc_Arc4Process */
 
+
+/*
+ * Testing wc_Init RsaKey()
+ */
+static int test_wc_InitRsaKey (void)
+{
+#ifndef NO_RSA
+    RsaKey  key[1];
+    int     ret;
+
+    printf(testingFmt, "wc_InitRsaKey()");
+
+    ret = wc_InitRsaKey(key, NULL);
+
+    /* Test bad args. */
+    if (ret == 0) {
+        ret = wc_InitRsaKey(NULL, NULL);
+        #ifndef HAVE_USER_RSA
+            if (ret == BAD_FUNC_ARG) {
+                ret = 0;
+            } else {
+        #else
+            if (ret == -101) {
+                ret = 0;
+            } else {
+        #endif
+            ret = SSL_FATAL_ERROR;
+        }
+    } /* end if */
+
+    if (wc_FreeRsaKey(key) || ret != 0) {
+        ret = SSL_FATAL_ERROR;
+    }
+
+    printf(resultFmt, ret == 0 ? passed : failed);
+
+#endif
+    return 0;
+} /* END test_wc_InitRsaKey */
+
+
+/*
+ * Testing wc_RsaPrivateKeyDecode()
+ */
+static int test_wc_RsaPrivateKeyDecode (void)
+{
+#if !defined(NO_RSA) && (defined(USE_CERT_BUFFERS_1024)\
+        || defined(USE_CERT_BUFFERS_2048)) && !defined(HAVE_FIPS)
+    RsaKey  key;
+    byte*   tmp;
+    word32  idx = 0;
+    int     ret = 0;
+    int     bytes = 0;
+
+    printf(testingFmt, "wc_RsaPrivateKeyDecode()");
+
+    tmp = (byte*)XMALLOC(4096, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    if (tmp == NULL) {
+        ret = SSL_FATAL_ERROR;
+    }
+    if (ret == 0) {
+        ret = wc_InitRsaKey(&key, NULL);
+    }
+    if (ret == 0) {
+        #ifdef USE_CERT_BUFFERS_1024
+            XMEMCPY(tmp, client_key_der_1024, sizeof_client_key_der_1024);
+            bytes = sizeof_client_key_der_1024;
+        #else
+            XMEMCPY(tmp, client_key_der_2048, sizeof_client_key_der_2048);
+            bytes = sizeof_client_key_der_2048;
+        #endif /* Use cert buffers. */
+
+        ret = wc_RsaPrivateKeyDecode(tmp, &idx, &key, (word32)bytes);
+    }
+    #ifndef HAVE_USER_RSA
+        /* Test bad args. */
+        if (ret == 0) {
+            ret = wc_RsaPrivateKeyDecode(NULL, &idx, &key, (word32)bytes);
+            if (ret == ASN_PARSE_E) {
+                ret = wc_RsaPrivateKeyDecode(tmp, NULL, &key, (word32)bytes);
+            }
+            if (ret == BAD_FUNC_ARG) {
+                ret = wc_RsaPrivateKeyDecode(tmp, &idx, NULL, (word32)bytes);
+            }
+            if (ret == ASN_PARSE_E) {
+                ret = 0;
+            } else {
+                ret = SSL_FATAL_ERROR;
+            }
+        }
+    #else
+        /* Test bad args. User RSA. */
+        if (ret == 0) {
+            ret = wc_RsaPrivateKeyDecode(NULL, &idx, &key, (word32)bytes);
+            if (ret == -101) {
```

Comment:
-101 should be USER_CRYPTO_ERROR here.  Please apply to all locations applicable.